### DISCUSSION TITLE
Gitflow without tags or commit messages?

### DISCUSSION THREAD

**Discussion by folbo** [GitTools/GitVersion#4072] [Q&A]
Hey, I have a problem configuring GitVersion the way that wouldn't require my teammates to create tags on main :-) Not sure if it's even possible (however I can't see the reason why not).

We work in rather simple GitFlow strategy, so we branch our `features` off the `develop`, squash-merge them back to `develop` and when the fine amount of features is collected on `develop` we do a normal-merge -no-ff to `main`. Sometimes we do hotfixes on main but it's a rare situation.

GitVersion on develop works like charm, I'm getting 1.1.0-alpha1, 1.1.0-alpha2, 1.1.0-alpha3 and so on for each squash-merged feature.
However, when develop is merged to main:
- `main` gets version 1.0.1
- further commits on `develop` get versions 1.1.0-alpha4, 1.1.0-alpha5, 1.1.0-alpha6

What I expect after merge is:
- `main` gets version 1.1.0 (becaue the prerelease tag should just be removed from the version being prepared on develop)
- further commits on `develop` get versions 1.2.0-alpha1, 1.2.0-alpha2, 1.2.0-alpha3 (because version is bumped after merge to main.

Of course setting a v1.1.0 tag on the merge commit helps but it requires some manual work which my colleagues aren't happy with. Any ideas how can I achieve my expectations using GitVersion?

**Comment by HHobeck** [comment]
You need to do one of the following points:
- use a no fast forward merge with an appropriated commit message from `develop` to `main`
- use the bump message feature and extend the commit message with e.g. `+semver: Minor`
- tag one of your pre-release commits on the `develop` branch with e.g. `1.1.0-alpha.6`
